src/Channel.cpp

```diff
@@ -145,22 +145,22 @@ void Channel::topic(Client *client, std::string &topic)
 			client->send_to_Client(name + ":" + ch_topic);
 		return ;
 	}
-	if (mode->isTopic() && !client_map[client]->isOperate())
+	if (!client_map[client]->isOperate()) // ch_mode->isTopic 필요
 		throw IRCException(":You must be a channel half-operator");
 	this->ch_topic = topic;
 	broadcast("332: " + ch_topic);
 }
 
 void Channel::mode(Client *client, std::vector<std::string> mode_vect)
 {
-	if (mode_str.empty())
+	if (mode_vect.empty())
 	{
-		client->send_to_Client("324" + ch_mode.toString());
+		client->send_to_Client("324"); // ch_mode->toString()
```

Comment:
std::map<C*, CM*>::iterator it = client_map.find(client);
bool isJoined = true;
if (it == client_map.end())
    isJoined = false;
client->send_to_Client("324" + ch_mode->getMode(isJoined))